cmd/protoc-gen-alias/main.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package main
+
+import (
+	"path/filepath"
+	"strings"
+
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func main() {
+	protogen.Options{}.Run(func(gen *protogen.Plugin) error {
+		for _, f := range gen.Files {
+			if !f.Generate {
+				continue
+			}
+			generateFile(gen, f)
+		}
+		return nil
+	})
+}
+
+func generateFile(gen *protogen.Plugin, file *protogen.File) {
+	ourVersion := filepath.Base(filepath.Dir(file.Desc.Path()))
+	var versions []string
+	for _, msg := range file.Messages {
+		for _, line := range strings.Split(msg.Comments.Leading.String(), "\n") {
+			if strings.HasPrefix(line, "// +cue-gen:") {
+				items := strings.Split(line, ":")
+				if items[2] == "versions" {
+					for _, v := range strings.Split(items[3], ",") {
```

Comment:
same here about index out of guarding; maybe just exist early if strings.Split returns too short a length